apps/emqx_authn/src/simple_authn/emqx_authn_redis.erl

```diff
@@ -141,16 +141,22 @@ authenticate(
                 {ok, []} ->
                     ignore;
                 {ok, Values} ->
-                    Selected = merge(Fields, Values),
-                    case
-                        emqx_authn_utils:check_password_from_selected_map(
-                            Algorithm, Selected, Password
-                        )
-                    of
-                        ok ->
-                            {ok, emqx_authn_utils:is_superuser(Selected)};
-                        {error, _Reason} ->
-                            ignore
+                    case check_query_matched(Values) of
+                        false ->
+                            %% key not exists
+                            ignore;
+                        _ ->
+                            Selected = merge(Fields, Values),
+                            case
+                                emqx_authn_utils:check_password_from_selected_map(
```

Comment:
I'd like to see some justification of this clause:
https://github.com/emqx/emqx/blob/master/apps/emqx_authn/src/emqx_authn_utils.erl#L89-L91

Reply:
worst case: maybe add a (2nd) clause to `emqx_authn_utils:check_password_from_selected_map` like this:
```
check_password_from_selected_map(_Algo, Selected, _Password) when Selected =:= #{} ->
  ignore;
```

Reply:
ok. it seems mysql and pgsql both had the empty query result handled locally.
maybe it makes sense to keep this logically then.